pkg/specs/secrets.go

```diff
@@ -51,3 +53,21 @@ func LoadFromSecretMatchingLabel(client kubernetes.Interface, labelSelector stri
 
 	return secretsMatchingKey, nil
 }
+
+func GetSecretMatchingLabel(client kubernetes.Interface, labelSelector string, namespace string, key string) (*corev1.Secret, error) {
+
+	secrets, err := client.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range secrets.Items {
+		_, ok := secret.Data[key]
+		if !ok {
+			continue
+		}
+		return &secret, nil
```

Comment:
How is this function intended to be used? This will just return the first secret found matching the label correct? Is that what we intend?